Archive/Draft/RFCNNNN-PowerShell-LocalSession-Configuration.md

```diff
@@ -0,0 +1,192 @@
+
+The PowerShell executable, `pwsh.exe`, can be started in 'server mode' by including the non-public `-S` command line switch.
+In this mode, PowerShell does not present a console window and prompt for command line input.
+Instead, it runs a listener on the process standard input stream, listening for remoting protocol (PSRP) messages, and sending protocol message responses on the process standard output stream.
+In this way PowerShell functions as a remoting endpoint for PowerShell remoting connections.
+PowerShell's `job`, `named pipe`, and `SSH` remoting connections all use PowerShell in server mode as an endpoint to connect to.
+In addition, the new [Custom Remote Connections feature](https://github.com/PowerShell/PowerShell-RFC/blob/master/Archive/Draft/RFC0063-Custom-Remote-Connections.md), will also use PowerShell in server mode for a remoting endpoint.  
+
+However, PowerShell in server mode has no way to configure the session and currently runs in its normal 'default' configuration.  
+
+### PowerShell `-ConfigurationFile` Command Line Switch
+
+The main proposal of this RFC is to add a new command line switch to `pwsh.exe` that takes an optional file path string to a PowerShell configuration (.pssc) file.
+This file would then be used to configure the default session runspace with the contained configuration information.
+The new command line switch would make configuring PowerShell running in server mode very easy.  
+
+For example, a PowerShell SSH connection endpoint is created by setting up a SSH 'subsystem' that runs `pwsh.exe`.
+Multiple PowerShell remoting endpoints could be created by using the new `-ConfigurationFile` command line switch:
+
+sshd_config file:
+
+```powershell
+...
+# override default of no subsystems
+#Subsystem  sftp        sftp-server
+# PowerShell endpoint configurations
+Subsystem   powershell1 /opt/microsoft/powershell/7/pwsh -S -NOLOGO -ConfigurationFile /etc/ssh/PSConfig1.pssc
+Subsystem   powershell2 /opt/microsoft/powershell/7/pwsh -S -NOLOGO -ConfigurationFile /etc/ssh/PSConfig2.pssc
+...
+```
+
+### Windows/WinRM Configuration Elements
+
+Unfortunately, some of the configuration elements supported in a .pssc configuration file is for Windows/WinRM remote connections only.
+For example, `virtual` and `group managed` accounts are Windows platform only.
+And JEA role definitions are based on Windows platform accounts and groups.  
+
+We want to separate Windows/WinRM specific configuration elements from what PowerShell supports implicitly.
+The new `-ConfigurationFile` command line switch and policy setting will support PowerShell configurations only, and will throw a terminating error if unsupported WinRM/Windows configurations are specified.  
+
+The following are Windows/WinRM configuration elements that will **not** be supported for local sessions:
+
+```powershell
+RunAsVirtualAccount
+RunAsVirtualAccountGroups
+GroupManagedServiceAccount
+RoleDefinitions
+RequiredGroups
+PowerShellVersion
```

Comment:
Perhaps it makes sense to support `PowerShellVersion` specially in server mode.

Reply:
This RFC is just for local configuration and doesn't affect server mode.

Reply:
Lines 97 and 103 seem to indicate this is all about server mode??